std/typecons.d

```diff
@@ -3275,6 +3288,50 @@ auto nullable(T)(T t)
 
     assert(foo.approxEqual(bar));
 }
+// bugzilla issue 19037
+@safe unittest
+{
+    import std.datetime : SysTime;
+
+    struct Test
+    {
+        bool b;
+        @disable this();
+        this(bool b) { this.b = b; }
+        nothrow invariant { assert(b == true); }
+        void opAssign(Test test2) @safe { this.b = test2.b; this._st = test2._st; }
+        ~this() @safe { }
+        SysTime _st;
+    }
+    Nullable!Test nt;
+    // mustn't call opAssign on Test.init in nt, because the invariant will be called
+    // on the Test.init that has to be in Nullable (because Test() is disabled),
+    // and Test.init violates its invariant.
+    nt = Test(true);
+    // destroy value
+    nt.nullify;
+    // don't run destructor on T.init in Nullable!
```

Comment:
You don't test for this

Reply:
The invariant effectively "tests" for it. I'll add a comment.